cmd/mikrotik-codegen/internal/codegen/templates.go

```diff
@@ -0,0 +1,189 @@
+package codegen
+
+const (
+	generatedNotice = "// This code was generated. Review it carefully."
+
+	resourceDefinitionTemplate = `
+package {{ .Package }}
+
+import (
+	{{ range $import := .Imports -}}
+		"{{ $import }}"
+	{{ end }}
+	tftypes "github.com/hashicorp/terraform-plugin-framework/types"
+
+)
+{{ $resourceStructName := .ResourceName | firstLower}}
+type {{$resourceStructName}} struct {
+	client *client.Mikrotik
+}
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &{{$resourceStructName}}{}
+	_ resource.ResourceWithConfigure   = &{{$resourceStructName}}{}
+	_ resource.ResourceWithImportState = &{{$resourceStructName}}{}
+)
+
+// New{{.ResourceName}}Resource is a helper function to simplify the provider implementation.
+func New{{.ResourceName}}Resource() resource.Resource {
+	return &{{$resourceStructName}}{}
+}
+
+func (r *{{$resourceStructName}}) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*client.Mikrotik)
+}
+
+// Metadata returns the resource type name.
+func (r *{{$resourceStructName}}) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_{{.ResourceName | lowercase}}"
+}
+
+// Schema defines the schema for the resource.
+func (s *{{$resourceStructName}}) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Creates a MikroTik {{.ResourceName}}.",
+		Attributes: map[string]schema.Attribute{
+			{{range .Fields -}}
+			"{{.AttributeName}}": schema.{{.Type.Name}}Attribute{
+				Required: {{.Required}},
+				Optional: {{.Optional}},
+				Computed: {{.Computed}},
+				{{if .Computed -}}
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
```

Comment:
The usages for "string" in the struct and package name should be updated to use `.Type.Name`.

Reply:
done